packages/flutter/lib/src/services/image_resource.dart

```diff
@@ -16,13 +16,14 @@ typedef void ImageListener(ui.Image image);
 /// image object might change over time, either because the image is animating
 /// or because the underlying image resource was mutated.
 class ImageResource {
```

Comment:
Essentially, it's exactly that. a Future around a ui.Image and its scale. What it adds on top of that is the ability to unregister a listener, and the error handling. The unregistering in particular is critical for how the framework works. Eventually we may also extend it to handle animated GIFs or push-JPEGs.